src/Common/src/System/Collections/Generic/LowLevelDictionary.IDictionary.cs

```diff
@@ -0,0 +1,149 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Collections.Generic
+{
+    internal partial class LowLevelDictionary<TKey, TValue> : IDictionary
+    {
+        bool IDictionary.IsFixedSize => false;
+        bool IDictionary.IsReadOnly => false;
+
+        object IDictionary.this[object key]
+        {
+            get { return this[(TKey)key]; }
```

Comment:
Instead of throwing `KeyNotFoundException`, this should return null if the key isn't found.